app.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require 'sinatra/base'
+require 'json'
+require 'haversine'
+require 'bigdecimal'
+require_relative 'lib/person_dto'
+require_relative 'lib/data_handlers/json_handler'
+require_relative 'lib/exceptions/data_unavailable_error'
+
+# Main entry point for routes
+class App < Sinatra::Base
+  
+  #sanitise user input
+  before do
+    params.each do |p, v|
+      params[p] = Rack::Utils.escape_html(v)
+    end
+  end
+  
+  # display all
+  get '/' do
+    # set defaults for bristol
+    @form_values = {params['calcAvgValue'] => 'off',
+                     'lat' => '51.450167', 'lon' => '-2.594678'}
+
+    json_handler = JsonHandler.new
+    begin
+      @people_stats = json_handler.get_person_dtos(true)
+    rescue DataUnavailableError => error
+      halt 400, error.message
+    end
+    status 200
+    erb :index
+  end
+
+  # filter and calculate average value depending on user input
+  post '/' do
+    @people_stats = {}
+    @form_values = {'lon'          => params['lon'],
+                    'lat'          => params['lat'],
+                    'radius'       => params['radius'],
+                    'unit'         => params['unit'],
+                    'calcAvgValue' => params['calcAvgValue']}
+
+    json_handler = JsonHandler.new
+    begin
+      person_dtos = json_handler.get_person_dtos(true)
+    rescue DataUnavailableError => error
+      halt 400, error.message
+    end
+
+    @people_stats = sorted_data_within_radius(person_dtos, true)
+
+    @avg_value = calculate_average_value(@people_stats) if params['calcAvgValue'] == 'on'
+    status 200
+    erb :index
+  end
+
+  private
+
+  def within_radius?(lat, lon, radius, distance_unit, person)
```

Comment:
I would move any distance calculation code into a isolated class with a single responsibility for calculating distances.